From: Jiri Kosina <jkosina@suse.cz>
Subject: x86/retpoline: don't perform thunk calls in ring3 vsyscall code
Patch-mainline: Never, SUSE-specific
References: bsc#1085331

Do not generate calls to retpoline thunks from vsyscall code which is running
in ring3 with user CR3, as the in-kernel thunks are obviously unreachable in
that environment (and accessing them causes #PF), and the retpoline protection
doesn't bring anything there (it's definitely not protecting the kernel).

Signed-off-by: Jiri Kosina <jkosina@suse.cz>
---
 arch/x86/kernel/Makefile |    2 ++
 1 file changed, 2 insertions(+)

--- a/arch/x86/kernel/Makefile
+++ b/arch/x86/kernel/Makefile
@@ -35,6 +35,8 @@ GCOV_PROFILE_paravirt.o		:= n
 # vread_tsc_64 is hot and should be fully optimized:
 CFLAGS_REMOVE_vread_tsc_64.o = -pg -fno-optimize-sibling-calls
 
+CFLAGS_REMOVE_vsyscall_64.o = $(RETPOLINE_CFLAGS)
+
 obj-y			:= process_$(BITS).o signal.o entry_$(BITS).o
 obj-y			+= traps.o irq.o irq_$(BITS).o dumpstack_$(BITS).o
 obj-y			+= time.o ioport.o ldt.o dumpstack.o
